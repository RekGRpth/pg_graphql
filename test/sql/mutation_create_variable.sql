begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );

    select graphql.resolve($$
    mutation newAccount($emailAddress: String) {
       xyz: createAccount(object: {
        email: $emailAddress
      }) {
        id
      }
    }
    $$,
    variables := '{"emailAddress": "foo@bar.com"}'::jsonb
    );


    select graphql.resolve($$
    mutation newAccount($acc: AccountInsertInput) {
       createAccount(object: $acc) {
        id
      }
    }
    $$,
    variables := '{"acc": {"email": "bar@foo.com"}}'::jsonb
    );

    -- Should fail with field does not exist
    select graphql.resolve($$
    mutation createAccount($acc: AccountInsertInput) {
       createAccount(object: $acc) {
        id
      }
    }
    $$,
    variables := '{"acc": {"doesNotExist": "other"}}'::jsonb
    );

rollback;
