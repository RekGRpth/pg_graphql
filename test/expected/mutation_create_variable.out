begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    select graphql.resolve($$
    mutation newAccount($emailAddress: String) {
       xyz: createAccount(object: {
        email: $emailAddress
      }) {
        id
      }
    }
    $$,
    variables := '{"emailAddress": "foo@bar.com"}'::jsonb
    );
                  resolve                   
--------------------------------------------
 {"data": {"xyz": {"id": 1}}, "errors": []}
(1 row)

    select graphql.resolve($$
    mutation newAccount($acc: AccountInsertInput) {
       createAccount(object: $acc) {
        id
      }
    }
    $$,
    variables := '{"acc": {"email": "bar@foo.com"}}'::jsonb
    );
                       resolve                        
------------------------------------------------------
 {"data": {"createAccount": {"id": 2}}, "errors": []}
(1 row)

    -- Should fail with field does not exist
    select graphql.resolve($$
    mutation createAccount($acc: AccountInsertInput) {
       createAccount(object: $acc) {
        id
      }
    }
    $$,
    variables := '{"acc": {"doesNotExist": "other"}}'::jsonb
    );
                                   resolve                                    
------------------------------------------------------------------------------
 {"data": null, "errors": ["Unknown field 'doesNotExist' on type 'Account'"]}
(1 row)

rollback;
